cvat-ui/src/reducers/annotation-reducer.ts

```diff
@@ -353,6 +358,7 @@ export default (state = defaultState, action: AnyAction): AnnotationState => {
                 annotations: {
                     ...state.annotations,
                     activatedStateID: null,
+                    unsaved: true,
```

Comment:
There are a lot of ways to change annotations. Do not try to compute whether annotations was updated or not here. 
``cvat-core`` has a method hasUnsavedChanges(), please, use it instead. 

Like: ``jobInstance.annotations.hasUnsavedChanges()`` // true or false